recipe/meta.yaml

```
@@ -1,64 +1,90 @@
 {% set name = "datasketches" %}
-{% set version = "4.1.0" %}
+{% set version = "5.2.0" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 6a57e4d1e93ccf1035604e3d9181dbe490281673833ce5be477c197b1195c840
-  patches:                                     # [win]
-  - patches/remove-generator-on-windows.patch  # [win]
+  - url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+    sha256: c00d61da4695e00036e63f590999f584cc39246cbb147b171f375f792604a612
+    patches:
+      - patches/0001-add-ninja-generator.patch  # [win]
+      - patches/0002-use-local-datasketches-cpp.patch
+      - patches/0003-fix-osx-linker.patch  # [osx]
+      - patches/0004-prevent-adding-debug-symbol-in-the-_datasketches-nam.patch  # [win]
+    folder: datasketches
+  - url: https://github.com/apache/{{ name }}-cpp/archive/refs/tags/{{ version }}.tar.gz
+    sha256: 63e6bda660ee9730cc39746de87fd983118c28149e32ff9389fac98b27010a53
+    folder: datasketches-cpp
 
 build:
   number: 0
-  script: {{ PYTHON }} -m pip install . -vv --no-build-isolation --no-deps
+  script: |
+    cd datasketches
+    {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  skip: True  # [py<38]
   missing_dso_whitelist:  # [s390x]
     - "$RPATH/ld64.so.1"  # [s390x]
+  ignore_run_exports:
+    - numpy
 
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    # git for clone of catch2-populate
-    - git  # [linux]
-    - cmake
-    - make      # [not win]
+    - cmake >=3.16
+    - make  # [not win]
+    - patch  # [not win]
     - m2-patch  # [win]
   host:
     - python
     - pip
-    - pybind11 >=2.6.0
-    - setuptools
+    - numpy {{ numpy }}
+    - nanobind >=1.6
+    - setuptools >=30.3.0
     - wheel
   run:
     - python
-    - numpy
+    - numpy {{ numpy }}
```

```suggestion
    - numpy # bounds through run_exports.
```
Using {{ numpy }} in run forces numpy to a specific version. It is better to rely on numpy's run_exports.